### PR TITLE
feat(dynamo): handle accumulate kwargs ("func", "initial")

### PR DESCRIPTION
**Pull request by jon-chuang**
Follow up to: https://github.com/pytorch/pytorch/pull/110683

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng @ezyang

### CODE REVIEW

File: torch/_dynamo/variables/misc.py [hunk 1]
```diff
@@ -790,20 +790,36 @@ def call_function(
...
             )
-        elif self.value is itertools.accumulate and not kwargs:
+        elif self.value is itertools.accumulate:
             from .builtin import BuiltinVariable
 
+            if any(key not in ["initial", "func"] for key in kwargs.keys()):
+                raise unimplemented("Unsupported kwarg for itertools.accumulate")
+
+            acc = None
+            if "initial" in kwargs:
```
Comment (jon-chuang) [inline torch/_dynamo/variables/misc.py:800]: acc = kwargs.get("initial")

**Comment by jon-chuang** [comment]
Ready for review @voznesenskym @ezyang

File: torch/_dynamo/variables/misc.py [hunk 2]
```diff
@@ -790,20 +790,34 @@ def call_function(
             return variables.ListIteratorVariable(
                 items, mutable_local=MutableLocal(), **options
             )
-        elif self.value is itertools.accumulate and not kwargs:
+        elif self.value is itertools.accumulate:
             from .builtin import BuiltinVariable
 
+            if any(key not in ["initial", "func"] for key in kwargs.keys()):
+                raise unimplemented("Unsupported kwarg for itertools.accumulate")
```
Comment (ezyang) [inline torch/_dynamo/variables/misc.py:797]: Nit: print the unsupported kwarg name in message

Comment (ezyang) [reply torch/_dynamo/variables/misc.py:797]: Also you don't raise unimplemented, just call it. (There is another site below that makes the same mistake)

**Review by ezyang — APPROVED** [review]

File: torch/_dynamo/variables/misc.py (hunk 2 above)
Comment (jon-chuang) [reply torch/_dynamo/variables/misc.py]: fixed

**Pull request closed without merging**